api/services/inventory/sock_store.go

```diff
@@ -204,3 +204,62 @@ func (s *SockStore) GetSockByID(sockID int) (*types.Sock, error) {
 
 	return &sock, nil
 }
+
+// UpdateSock updates an existing sock and its variants in the database
+func (s *SockStore) UpdateSock(sockID int, sock types.Sock, variants []types.SockVariant) error {
+	// Update the sock details
+	_, err := s.db.Exec(`
+		UPDATE socks
+		SET name = $1, description = $2, preview_image_url = $3
+		WHERE sock_id = $4 AND is_deleted = false`,
+		sock.Name, sock.Description, sock.PreviewImageURL, sockID)
+
+	if err != nil {
+		return fmt.Errorf("failed to update sock: %w", err)
+	}
+
+	// Update or insert each variant
+	for _, variant := range variants {
+		// Check if the variant exists
+		exists, err := s.sockVariantExists(sockID, variant.Size)
+		if err != nil {
+			return err
+		}
+
+		if exists {
+			// Update existing variant
+			_, err := s.db.Exec(`
+				UPDATE sock_variants
+				SET price = $1, quantity = $2
+				WHERE sock_id = $3 AND size = $4`,
+				variant.Price, variant.Quantity, sockID, variant.Size)
+
+			if err != nil {
+				return fmt.Errorf("failed to update variant: %w", err)
+			}
+		} else {
+			// Insert new variant
+			_, err := s.db.Exec(`
+				INSERT INTO sock_variants (sock_id, price, quantity, size)
+				VALUES ($1, $2, $3, $4)`,
+				sockID, variant.Price, variant.Quantity, variant.Size)
+
+			if err != nil {
+				return fmt.Errorf("failed to insert variant: %w", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// sockVariantExists checks if a sock variant exists for the given sock ID and size
+func (s *SockStore) sockVariantExists(sockID int, size string) (bool, error) {
```

Comment:
This is useful functionality, so worthwhile making into an public method for the store. 

Add it to the `SockStore` interface: `SockVariantExists(sockID int, size string) (bool, error)`

Reply:
Got it! Thank you